src/server/index.ts

```diff
@@ -331,33 +329,13 @@ export class Server<
 
     /**
      * Creates an elicitation request for the given parameters.
-     * @param params The parameters for the form elicitation request (explicit mode: 'form').
+     * For backwards compatibility, `mode` may be omitted for form requests and will default to `'form'`.
+     * @param params The parameters for the elicitation request.
      * @param options Optional request options.
      * @returns The result of the elicitation request.
      */
-    async elicitInput(params: ElicitRequestFormParams, options?: RequestOptions): Promise<ElicitResult>;
-    /**
-     * Creates an elicitation request for the given parameters.
-     * @param params The parameters for the URL elicitation request (with url and elicitationId).
-     * @param options Optional request options.
-     * @returns The result of the elicitation request.
-     */
-    async elicitInput(params: ElicitRequestURLParams, options?: RequestOptions): Promise<ElicitResult>;
-    /**
-     * Creates an elicitation request for the given parameters.
-     * @deprecated Use the overloads with explicit `mode: 'form' | 'url'` instead.
-     * @param params The parameters for the form elicitation request (legacy signature without mode).
-     * @param options Optional request options.
-     * @returns The result of the elicitation request.
-     */
-    async elicitInput(params: LegacyElicitRequestFormParams, options?: RequestOptions): Promise<ElicitResult>;
-
-    // Implementation (not visible to callers)
-    async elicitInput(
-        params: LegacyElicitRequestFormParams | ElicitRequestFormParams | ElicitRequestURLParams,
-        options?: RequestOptions
-    ): Promise<ElicitResult> {
-        const mode = ('mode' in params ? params.mode : 'form') as 'form' | 'url';
+    async elicitInput(params: ElicitRequestFormParams | ElicitRequestURLParams, options?: RequestOptions): Promise<ElicitResult> {
```

Comment:
Back to needing only 1 overload here. The possibility that `mode could be null/missing is handled inside the method.